apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerParameter.cs

```diff
@@ -160,7 +161,28 @@ internal object GetValidatedValue()
         /// </summary>
         /// <param name="options">Options to configure the <c>SpannerDbType</c> for this <c>SpannerParameter</c>.</param>
         /// <returns>The configured <see cref="SpannerDbType"/>.</returns>
-        internal SpannerDbType GetConfiguredSpannerDbType(SpannerConversionOptions options) => SpannerDbType;
+        internal SpannerDbType GetConfiguredSpannerDbType(SpannerConversionOptions options)
+        {
+            // Only if SpannerDbType of parameter is not explicitly provided by user.
+            if (_spannerDbType == null)
+            {
+                if (Value is decimal)
+                {
+                    // User needs to set the right dialect.
+                    if (options != null && options.UseSpannerNumericForDecimal)
+                    {
+                        return SpannerDbType.Numeric;
+                    }
+                    else if (options != null && options.UsePgNumericForDecimal)
+                    {
+                        return SpannerDbType.PgNumeric;
+                    }
+                }
+            }
+
+            // If we are here, use defaults.
+            return SpannerDbType;
```

Comment:
We definitely need to comment on the SpannerDbType property that it doesn't take the connection string into account.